src/game/client/components/mapimages.cpp

```diff
@@ -196,3 +147,76 @@ int CMapImages::GetOverlayCenter()
 {
 	return m_OverlayCenterTexture;
 }
+
+void CMapImages::SetTextureScale(int Scale)
+{
+	if (m_TextureScale == Scale)
+		return;
+
+	m_TextureScale = Scale;
+
+	if(Graphics() && m_OverlayCenterTexture != -1) // check if component was initialized
+	{
+		// reinitialize component
+		Graphics()->UnloadTexture(m_OverlayBottomTexture);
+		Graphics()->UnloadTexture(m_OverlayTopTexture);
+		Graphics()->UnloadTexture(m_OverlayCenterTexture);
+		m_OverlayBottomTexture = m_OverlayTopTexture = m_OverlayCenterTexture = -1;
+
+		InitOverlayTextures();
+	}
+}
+
+int CMapImages::GetTextureScale()
+{
+	return m_TextureScale;
+}
+
+int CMapImages::UploadEntityLayerText(int TextureSize, int YOffset)
+{	
+	void* pMem = calloc(1024 * 1024, 1);
+	int TextureID = Graphics()->LoadTextureRaw(1024, 1024, CImageInfo::FORMAT_ALPHA, pMem, CImageInfo::FORMAT_ALPHA, IGraphics::TEXLOAD_NOMIPMAPS);
+	free(pMem);
+
+	char buff[4];
+	str_format(buff, 4, "%d", 255);
+	int len = str_length(buff);
+
+	int FontSize = TextRender()->AdjustFontSize(buff, len, TextureSize);
+	YOffset += ((TextureSize - FontSize)/2);
+
+	for (int i = 0; i < 256; ++i)
+	{
+		str_format(buff, 4, "%d", i);
+		int len = str_length(buff);
+
+		float x = (i%16)*64;
+		float y = (i/16)*64;
+		
+		TextRender()->UploadEntityLayerText(TextureID, buff, len, x, y + YOffset, FontSize);
+	}
+
+	return TextureID;
+}
+
+
+void CMapImages::InitOverlayTextures()
+{
+	int TextureSize = 64*m_TextureScale/100;
+	int TextureToVerticalCenterOffset = (64-TextureSize)/2; // should be used to move texture to the center of 64 pixels area
+	
+	if(m_OverlayBottomTexture == -1)
+	{
+		m_OverlayBottomTexture = UploadEntityLayerText(TextureSize/2, 32+TextureToVerticalCenterOffset/2);
+	}
+
+	if(m_OverlayTopTexture == -1)
+	{
+		m_OverlayTopTexture = UploadEntityLayerText(TextureSize/2, TextureToVerticalCenterOffset/2);
+	}
+
+	if(m_OverlayCenterTexture == -1)
+	{
+		m_OverlayCenterTexture = UploadEntityLayerText(TextureSize, TextureToVerticalCenterOffset);
+	}
+}
```

Comment:
```suggestion
}

```
You should configure your editor to make sure it adds a newline at EOF.